File: src/main/java/org/reflections/scanners/ResourcesScanner.java
package org.reflections.scanners;

import org.reflections.vfs.Vfs.File;

/**
 * collects all resources that are not classes in a collection
 * <p>key: value - {web.xml: WEB-INF/web.xml}
 */
public class ResourcesScanner extends AbstractScanner {

    @Override
    public boolean acceptsInput(String file) {
        return !file.endsWith(".class"); //not a class
    }

    @Override
    public Object scan(File file, Object classObject) {
        getStore().put(file.getName(), file.getRelativePath());
        return classObject;
    }

    @Override
    public void scan(Object cls) {
        throw new UnsupportedOperationException(); //shouldn't get here
    }
}

File: src/main/java/org/reflections/scanners/ResourcesScanner.kt
package org.reflections.scanners

import org.reflections.adapters.ClassAdapter
import org.reflections.vfs.Vfs.File

/**
 * collects all resources that are not classes in a collection
 *
 * key: value - {web.xml: WEB-INF/web.xml}
 */
class ResourcesScanner : AbstractScanner() {

    override fun acceptsInput(file: String): Boolean {
        return !file.endsWith(".class") //not a class
    }

    override fun scan(file: File, classObject: ClassAdapter?): ClassAdapter? {
        store!!.put(file.name, file.relativePath!!)
        return classObject
    }

    override fun scan(cls: ClassAdapter) {
        throw UnsupportedOperationException() //shouldn't get here
    }
}
